rotkehlchen/tests/exchanges/test_bitstamp.py

```diff
@@ -0,0 +1,724 @@
+        return MockResponse(HTTPStatus.OK, '{"link_available": "1.00000000"}')
+
+    with patch.object(mock_bitstamp, '_api_query', side_effect=mock_api_query_response):
+        assert mock_bitstamp.query_balances() == ({}, '')
+
+
+def test_query_balances__skipped_not_asset_entry(mock_bitstamp):
+    """Test an entry that can't instantiate Asset is skipped
+    """
+    def mock_api_query_response(endpoint):  # pylint: disable=unused-argument
+        return MockResponse(HTTPStatus.OK, '{"bbbrrrlink_balance": "1.00000000"}')
+
+    with patch.object(mock_bitstamp, '_api_query', side_effect=mock_api_query_response):
+        assert mock_bitstamp.query_balances() == ({}, '')
+
+
+def test_query_balances__skips_inquirer_error(mock_bitstamp):
+    """Test an entry that can't get its USD price because of a remote error is
+    skipped
+    """
+    inquirer = MagicMock()
+    inquirer.find_usd_price.side_effect = RemoteError('test')
+
+    def mock_api_query_response(endpoint):  # pylint: disable=unused-argument
+        return MockResponse(HTTPStatus.OK, '{"link_balance": "1.00000000"}')
+
+    with patch('rotkehlchen.exchanges.bitstamp.Inquirer', return_value=inquirer):
+        with patch.object(mock_bitstamp, '_api_query', side_effect=mock_api_query_response):
+            assert mock_bitstamp.query_balances() == ({}, '')
+
+
+@pytest.mark.parametrize('should_mock_current_price_queries', [True])
+def test_query_balances__asset_balance(mock_bitstamp, inquirer):  # pylint: disable=unused-argument
+    """Test an entry that can't get its USD price is skipped
+    """
+    balances_data = (
+        """
+        {
+            "eth_available": "0.00000000",
+            "eth_balance": "32.00000000",
+            "eth_reserved": "0.00000000",
+            "eth_withdrawal_fee": "0.04000000",
+            "link_available": "0.00000000",
+            "link_balance": "1000.00000000",
+            "link_reserved": "0.00000000",
+            "link_withdrawal_fee": "0.25000000",
+            "xrp_available": "0.00000000",
+            "xrp_balance": "0.00000000",
+            "xrp_reserved": "0.00000000",
+            "xrp_withdrawal_fee": "0.02000000"
+        }
+        """
+    )
+
+    def mock_api_query_response(endpoint):  # pylint: disable=unused-argument
+        return MockResponse(HTTPStatus.OK, balances_data)
+
+    with patch.object(mock_bitstamp, '_api_query', side_effect=mock_api_query_response):
+        asset_balance, msg = mock_bitstamp.query_balances()
+        assert asset_balance == {
+            Asset('ETH'): Balance(
+                amount=FVal('32'),
+                usd_value=FVal('48'),
+            ),
+            Asset('LINK'): Balance(
+                amount=FVal('1000'),
+                usd_value=FVal('1500'),
+            ),
+        }
+        assert msg == ''
+
+
+# method: `_deserialize_trade()`
+def test_deserialize_trade__buy(mock_bitstamp):
```

Comment:
here and below why double underscore after `trade`?

Reply:
I tend to use underscores when two of more tests share the first part of the title (testing the same function) but apply to different logic. It helps separating words/terms that can already contain a `_`. I can stop it :P

Reply:
It's not bad, but since we don't do it in the rest of the code it's better if you also don't do it. After all a consistent looking codebase is important.